### DISCUSSION TITLE
How to determine if a message was consumed and the consumer details ？

### DISCUSSION THREAD

**Discussion by tianshimoyi** [apache/pulsar#18342] [Q&A]
1. How to determine if a message was consumed and the consumer details ？
2. peek-messages cannot get consumed messages, but namespace has set message retention policy

**Comment by huxiyu** [comment]
oh my god！这么重要的特性一直没有答案。我也需要...

**Comment by huxiyu** [comment]
rocketmq默认dashboard都可以看。
2年了，Pulsar有消费结果查询了没？
